### DISCUSSION TITLE
How to skip specific pod volume to velero backup

### DISCUSSION THREAD

**Discussion by sai8465** [vmware-tanzu/velero#7907] [Community support Q&A]
For  single pod we mount two volumes, we need to skip specific (one volume) volume to velero backup and already its have one solution is to add annotations to pod to skip volume(kubectl -n YOUR_POD_NAMESPACE annotate pod/YOUR_POD_NAME backup.velero.io/backup-volumes=YOUR_VOLUME_NAME_1,YOUR_VOLUME_NAME_2), except this is there any other solution

**Comment by kaovilai** [comment]
You can opt out of volumes using annotations like so https://velero.io/docs/v1.13/file-system-backup/#using-the-opt-out-approach

```
kubectl -n YOUR_POD_NAMESPACE annotate pod/YOUR_POD_NAME backup.velero.io/backup-volumes-excludes=YOUR_VOLUME_NAME_1,YOUR_VOLUME_NAME_2,...

```

**Comment by sseago** [comment]
Opt in/Opt out isn't for skipping volumes, though -- that just swaps between fs-backup and snapshots  for the annotated volumes.

**Reply by kaovilai** [reply]
> It is possible to exclude volumes from being backed up using the backup.velero.io/backup-volumes-excludes annotation on the pod.

https://velero.io/docs/v1.13/file-system-backup/#using-the-opt-out-approach:~:text=It%20is%20possible%20to%20exclude%20volumes%20from%20being%20backed%20up%20using%20the%20backup.velero.io/backup%2Dvolumes%2Dexcludes%20annotation%20on%20the%20pod.

If fsbackup is the only thing enabled, then these annotations should according to docs, prevent volumes from being fs-backup

**Reply by sseago** [reply]
@kaovilai This is true -- Velero will attempt to snapshot but will fail as long as both of the below are true:
1) no volumesnapshotter is configured for the volume
2) volume is not CSI or CSI is not enabled

If the above are not both true, the other way for opt-out to skip the volume would be to set snapshotVolumes: false along with opt-out.

That being said, this isn't really the purpose of the opt-out annotation -- using volume policies to skip would probably be a better choice, although I guess that's not yet possible since we can't select on volume name in those policies at least not as of 1.14.
